### PR TITLE
Add error message for unsupported Windows versions

### PR DESCRIPTION
**Pull request by thomas-senechal**
## Delivery Quality Checklist

- [ ] **Breaking Changes in API:**
  Does this PR introduce breaking changes in the API?
    - If yes, have you considered making it backward compatible?
    - If backward compatibility is not considered, set the "breaking-change" label.

- [ ] **Changelog:**
  - [ ] For bugfix PR, set the "bugfix" label
  - [ ] If this change should not appear in changelog, use "ignore-for-changelog" label

- [ ] **Version Update Handling:**
  Have you ensured that the version update by user is handled correctly?

- [ ] **PR Dependency:**
  Does this PR depend on another PR?
    - If yes, is it necessary for the dependency to be released prior to merging this one?

- [ ] **Documentation:**
  - [ ] Are any necessary changes made to user-facing documentation?
  - [ ] Confirm that API documentation is updated with any relevant changes.
  - [ ] Check that README and other documentation files are accurate and current.

### CODE REVIEW

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: installer/windows/build_installer.py [hunk 1]
```diff
@@ -153,6 +153,14 @@ def create_wxs_file():
             AllowSameVersionUpgrades="yes"
         />
 
+        <Condition Message="Oops! Massa Station cannot be installed on your current Windows version ([VersionNT]).\nWe require Windows 10 or a newer version for compatibility. Please consider updating your operating system to enjoy Massa Station.">
+            <!--
+                We require Windows 10 or a newer version for compatibility. Windows 10 and 11 are version 603.
+                We don't want to allow installation on Server versions of Windows.
+            -->
+            <![CDATA[Installed OR (VersionNT >= 603 AND MsiNTProductType = 1)]]>
```
[the remark this answers is not in the document]
Comment (thomas-senechal) [reply installer/windows/build_installer.py:161]: https://learn.microsoft.com/en-us/troubleshoot/windows-client/application-management/versionnt-value-for-windows-10-server

Windows is a nightmare 😭

[the remark this answers is not in the document]
Comment (thomas-senechal) [reply installer/windows/build_installer.py:161]: And here they say 603 is for W8.1
https://learn.microsoft.com/en-us/windows/win32/msi/operating-system-property-values

[the remark this answers is not in the document]
Comment (thomas-senechal) [reply installer/windows/build_installer.py:161]: And in my installer logs, it says I'm at 603

[the remark this answers is not in the document]
Comment (thomas-senechal) [reply installer/windows/build_installer.py:161]: > The installer sets the VersionNT property to the version number for the operating system, undefined if the operating system is not Windows NT, Windows 2000, Windows XP, Windows Vista, Windows Server 2008, or Windows 7. The value is an integer: MajorVersion * 100 + MinorVersion

https://learn.microsoft.com/en-us/windows/win32/msi/versionnt

[the remark this answers is not in the document]
Comment (thomas-senechal) [reply installer/windows/build_installer.py:161]: https://stackoverflow.com/q/71647237 🫠

[the remark this answers is not in the document]
Comment (thomas-senechal) [reply installer/windows/build_installer.py]: image
Using Build number

**Review by gregLibert — APPROVED** [review]

**Review by pivilartisant — APPROVED** [review]

**Review by Thykof — APPROVED** [review]

**Comment by thomas-senechal** [comment]
Just for the record, to test this, you need either to have a Windows prior to W10, or, modify the `WINDOWSEDITION` ( https://github.com/massalabs/station/pull/1216/files#diff-40085a0c3f8a1d74e1ec5324db618bbced740670ce3324856c05f60f09b4245bR170 ) to a version after yours (or one that does not exists yet)

**Pull request merged**